tools/run_tests/performance/bq_upload_result.py

```diff
@@ -213,6 +215,30 @@ def _populate_metadata_from_file(scenario_result, test_metadata_file):
     scenario_result['metadata'] = metadata
 
 
+def _populate_node_metadata_from_file(scenario_result, node_info_file):
+    node_metadata = {'driver': {}, 'servers': [], 'clients': []}
+    _node_info_to_bq_node_metadata_key_map = {
+        'Name': 'name',
+        'PodIP': 'podIP',
+        'NodeName': 'nodeName',
+    }
+    with open(node_info_file, 'r') as f:
```

Comment:
Sorry I was not able to review the PR earlier, do we need to add a check to make sure that the inode_info_file is readable ?

Reply:
Hey, thanks for commenting! I assume using "with open():" would let it jump out of the flow(execute line 239) if an exception is met.

Reply:
I might understand this wrong, but I think the with open() helps you close the file no matter what, but if there is an exception (i.e. file non-existing), it will raise the exception. 
https://colab.sandbox.google.com/drive/1LdHGnqRIeHO5Qu1qh2WiXBPMPW8d4BBQ?resourcekey=0-MuDDSsaWE-zFIpN-wPEdIg